MobiusCore/Source/EffectHandlers/EffectRouterDSL.swift

```diff
@@ -0,0 +1,166 @@
+// Copyright (c) 2019 Spotify AB.
+//
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+public extension EffectRouter where Input: Equatable {
+    /// Route to an `EffectHandler` for all effects which are equal to `constant`.
+    /// - Parameter constant: the constant to use for comparisons
+    /// - Parameter handler: the effect handler which should handle `constant`
+    func routeConstant(
+        _ constant: Input,
+        toHandler handler: EffectHandler<Input, Output>
+    ) -> EffectRouter<Input, Output> {
+        return add(
+            path: { effect in effect == constant ? constant : nil },
+            to: handler
+        )
+    }
+
+    /// Route to an side-effecting function for all effects which are equal to `constant`.
+    /// - Parameter constant: the constant to use for comparisons
+    /// - Parameter fireAndForget: the function which should perform a side-effect
+    func routeConstant(
+        _ constant: Input,
+        toVoid fireAndForget: @escaping () -> Void
+    ) -> EffectRouter<Input, Output> {
+        return add(
+            path: { effect in effect == constant ? constant : nil },
+            to: EffectHandler(
+                handle: { _, _ in fireAndForget() },
+                disposable: AnonymousDisposable {}
+            )
+        )
+    }
+
+    /// Route to a function which returns an event for all effects which are equal to `constant`.
+    /// - Parameter constant: the constant to use for comparisons
+    /// - Parameter outputFunction: the function which returns an event for the `constant`
+    func routeConstant(
+        _ constant: Input,
+        toEvent outputFunction: @escaping () -> Output
```

Comment:
Is there a downside to making this `() -> Output?`?

Reply:
I'm not sure how that would work, given that it cannot be called right away

Reply:
I don’t see the connection. The handler would just be:
```swift
{ _, dispatch in 
    if let event = outputFunction() {
        dispatch(event)
    }
}
```

Reply:
Sorry, I thought the suggestion was to remove the `@escaping`. I see now. Yeah I think that makes sense.